WpfDesign.Designer/Project/Extensions/BorderForMouseOver.cs

```diff
@@ -31,18 +31,33 @@ namespace ICSharpCode.WpfDesign.Designer.Extensions
 	public class BorderForMouseOver : AdornerProvider
 	{
 		readonly AdornerPanel adornerPanel;
+		readonly Border border = new Border();
 
 		public BorderForMouseOver()
 		{
 			adornerPanel = new AdornerPanel();
 			adornerPanel.Order = AdornerOrder.Background;
 			this.Adorners.Add(adornerPanel);
-			var border = new Border();
 			border.BorderThickness = new Thickness(1);
 			border.BorderBrush = Brushes.DodgerBlue;
 			border.Margin = new Thickness(-2);
 			AdornerPanel.SetPlacement(border, AdornerPlacement.FillContent);
 			adornerPanel.Children.Add(border);
 		}
+
+		protected override void OnInitialized()
+		{
+			base.OnInitialized();
+
+			if (ExtendedItem.Component is Line line)
+			{
+				// To display border of Line in correct position.
+				border.Margin = new Thickness
```

Comment:
if the extended item isn't a line, don't you need to reset the margin?

Reply:
Yes, I don't think it is needed.

It is because instance of BorderForMouseOver is re-created when mouse over other item.